<launch>
  <!-- 加载参数 -->
  <arg name="config_file" default="$(find URDF_armv001_demo)/config/serial_config.yaml"/>
  <arg name="port" default="/dev/ttyUSB0"/>
  <arg name="baudrate" default="115200"/>
  <arg name="debug" default="true"/>
  
  <!-- 启动串口节点 -->
  <node name="serial_node" pkg="URDF_armv001_demo" type="serial_node.py" output="screen">
    <param name="config_file" value="$(arg config_file)"/>
    <param name="port" value="$(arg port)"/>
    <param name="baudrate" value="$(arg baudrate)"/>
    <param name="debug" value="$(arg debug)"/>
  </node>
  
  <!-- 如果debug参数为true，启动rqt_console查看详细日志 -->
  <node if="$(arg debug)" name="rqt_console" pkg="rqt_console" type="rqt_console"/>
</launch> 